src/github.com/getlantern/peerscanner/web.go

```diff
@@ -51,36 +56,46 @@ func startHttp() {
 // register is the entry point for peers registering themselves with the service.
 // If peers are successfully vetted, they'll be added to the DNS round robin.
 func register(resp http.ResponseWriter, req *http.Request) {
-	name, ip, port, err := getHostInfo(req)
-	if err == nil && port != 443 {
-		err = fmt.Errorf("Port %d not supported, only port 443 is supported", port)
+	name, ip, port, supportedFronts, err := getHostInfo(req)
+	if err == nil && !(port == "80" || port == "443") {
+		err = fmt.Errorf("Port %d not supported, only ports 80 and 443 are supported", port)
```

Comment:
I think checking domain fronting providers and the ratio of peers vs fallbacks could be done in a different service, the responsibility of which would be to update cloud.yaml (as opposed to the DNS/CDN registries, which this service would treat as read-only).

Either way, whoever checks fronting providers only needs test the round robin itself, to make sure that it can proxy through _some_ server via each fronting provider.  That would be way cheaper than peerscanner using domain fronting all the time, esp. if we switch to serving domain fronted requests through port 80.